WooCommerce/Classes/CardPresentPayments/CardPaymentsAdaptor.swift

```diff
@@ -0,0 +1,147 @@
+        let preflightController = await CardPresentPaymentPreflightController(
+            siteID: siteID,
+            configuration: CardPresentConfigurationLoader().configuration,
+            rootViewController: UIViewController(),
+            alertsPresenter: self,
+            onboardingPresenter: self,
+            bluetoothConnectionController: bluetoothConnectionController,
+            builtInConnectionController: tapToPayConnectionController)
+        let orderPaymentUseCase = await CollectOrderPaymentUseCase(siteID: siteID,
+                                                                   order: order,
+                                                                   formattedAmount: currencyFormatter.formatAmount(order.total, with: order.currency) ?? "",
+                                                                   // moved from EditableOrderViewModel.collectPayment(for: Order)
+                                                                   rootViewController: UIViewController(),
+                                                                   // We don't want to use this at all, but it's currently required by the existing code.
+                                                                   // TODO: replace `rootViewController` with a protocol containing the UIVC functions we need, and implement that here.
+                                                                   onboardingPresenter: self,
+                                                                   configuration: CardPresentConfigurationLoader().configuration,
+                                                                   alertsPresenter: self,
+                                                                   preflightController: preflightController)
+        return await withCheckedContinuation { continuation in
+            orderPaymentUseCase.collectPayment(using: discoveryMethod) { error in
+                if let error = error as? CardPaymentErrorProtocol {
+                    continuation.resume(returning: .failure(error))
+                } else {
+                    continuation.resume(returning: .failure(CardPaymentsAdaptorError.unknownPaymentError(underlyingError: error)))
+                }
+            } onCancel: {
+                continuation.resume(returning: .cancellation)
+            } onPaymentCompletion: {
+                // no-op – not used in PaymentMethodsViewModel anyway so this can be removed
+            } onCompleted: {
+                continuation.resume(returning: .success(order))
+            }
+        }
+    }
+
+    enum CardPaymentsAdaptorError: Error, CardPaymentErrorProtocol {
+        var retryApproach: CardPaymentRetryApproach {
+            .restart
+        }
+
+        case unknownPaymentError(underlyingError: Error)
+    }
+}
+
+extension CardPresentPaymentsAdaptor: CardPresentPaymentsOnboardingPresenting, CardPresentPaymentAlertsPresenting {
```

Comment:
Might be a naive question so feel free to correct: I think it'd be nice if the consumer only knows about the payments/reader-related interface like `collectPayment(for:using:)`, and not the functions/variables that come with these protocol conformance. We can also create a protocol for the adaptor to address this if possible.

Reply:
Yep, agreed. I didn't do it initially in the experiment but then did it to answer a P2 question, and it makes it clearer what the consumer should know about.

Having the adaptor implement a bunch of other protocols to do its work makes it harder to have clarity there, but the protocol helps.